# **Quelles sont les collectivités ?**

Nous nous intéressons ici à constituer un référentiel national le plus complet et actualisé des collectivités et autres organisations publiques et para-publiques.

Le point de départ de l'investigation excentrique est de lister toutes les questions du type, quelles sont :
* ma commune
* les communes limitrophes de ma commune
* la communauté de communes dont ma commune est membre
* les communautés de communes limitrophes
* le département et toutes les divisions territoriales intermédiaires dont est membre ma commune, ma communauté de commune, etc
* ...

Cette zone d'investigation est donc à croiser avec celle des élus qui permet de comprendre et rendre apparente la super-structure institutionnelle (administrative) et les incarnants des rôles et places qu'elle définit.

### [Service-public.fr - Annuaire de l’administration - Base de données locales](https://www.data.gouv.fr/fr/datasets/service-public-fr-annuaire-de-l-administration-base-de-donnees-locales/)

Information sur le jeu de données :
* **Editeur** : DILA
* **Mise à jour** : 8 décembre 2022
* **Licence** : [Licence Ouverte / Open Licence](https://www.etalab.gouv.fr/wp-content/uploads/2014/05/Licence_Ouverte.pdf)
* **Organisation** : [Premier ministre](https://www.data.gouv.fr/fr/organizations/premier-ministre/)
* **Source** :
    * http://lecomarquage.service-public.fr/donnees_locales_v2/
    * http://lecomarquage.service-public.fr/donnees_locales_v4/
* **API** : https://api.gouv.fr/les-api/api_etablissements_publics
* **Thématiques**
    * ...
* **Utilisation**
    * ... DL
    * .. réutilisations
    * .. abonnés
    * .. vues

La Base de données locales référence plus de 63 000 guichets publics locaux (mairies, organismes sociaux, services de l'état, etc.). Elle fournit leurs coordonnées (adresses, téléphones, site internet, horaires d'ouverture, coordonnées de géolocalisation). En complément, sont indexés plus de 36 000 fichiers des communes, précisant la compétence géographique des guichets.

La liste des types d'organismes référencés dans la base est consultable dans le fichier PDF joint ci-dessous.

Le schéma et sa documentation sont disponibles à l'adresse : Specifications-datagouv-base-de-donnees-locales-Service-public_V1.1.pdf

REMARQUE IMPORTANTE : Depuis le 4 juillet 2022, ce jeu de données a changé de format et de structure. Les données sont au format json.


In [11]:
import pandas as pd
import numpy as np
base = '../data/premier_ministre/données_locales/all_latest/'
filename = '2022-12-12_002751-data.gouv_local.json'
data = pd.read_json(base + filename)
jsons = data.service.values
data = pd.DataFrame.from_dict(pd.json_normalize(jsons), orient='columns')
data.index.names = ['service']
data = data.applymap(lambda x: np.nan if x == '' or x == [] else x)
data = data.dropna(axis=1, how='all')
display(data)

plage_ouverture  \
service                                                      
0                                                      NaN   
1        [{'nom_jour_debut': 'Mardi', 'nom_jour_fin': '...   
2                                                      NaN   
3        [{'nom_jour_debut': 'Lundi', 'nom_jour_fin': '...   
4        [{'nom_jour_debut': 'Jeudi', 'nom_jour_fin': '...   
...                                                    ...   
67509    [{'nom_jour_debut': 'Lundi', 'nom_jour_fin': '...   
67510                                                  NaN   
67511    [{'nom_jour_debut': 'Lundi', 'nom_jour_fin': '...   
67512    [{'nom_jour_debut': 'Lundi', 'nom_jour_fin': '...   
67513    [{'nom_jour_debut': 'Lundi', 'nom_jour_fin': '...   

                                             site_internet  \
service                                                      
0                                                      NaN   
1        [{'libelle': '', 'valeur': 'http://www.ville-f...   
2                                                      NaN   
3        [{'libelle': '', 'valeur': 'https://auvergne-r...   
4        [{'libelle': '', 'valeur': 'https://www.cohesi...   
...                                                    ...   
67509    [{'libelle': '', 'valeur': 'https://www.police...   
67510    [{'libelle': '', 'valeur': 'https://www.mas-as...   
67511    [{'libelle': '', 'valeur': 'http://www.mairie-...   
67512    [{'libelle': '', 'valeur': 'http://www.ville-l...   
67513                                                  NaN   

                                                 copyright siren  \
service                                                            
0                                  Ministère de la Justice   NaN   
1        Direction de l'information légale et administr...   NaN   
2                                  Ministère de la Justice   NaN   
3        Direction de l'information légale et administr...   NaN   
4        Direction de l'information légale et administr...   NaN   
...                                                    ...   ...   
67509    Direction de l'information légale et administr...   NaN   
67510                              Ministère de la Justice   NaN   
67511    Direction de l'information légale et administr...   NaN   
67512    Direction de l'information légale et administr...   NaN   
67513    Direction de l'information légale et administr...   NaN   

                     ancien_code_pivot texte_reference         partenaire  \
service                                                                     
0                        spip-64102-01             NaN  ministere_justice   
1                      mairie-77188-01             NaN                NaN   
2        permanence_juridique-45203-01             NaN  ministere_justice   
3               droit_travail-69266-01             NaN                NaN   
4                        msap-34182-01             NaN     france_service   
...                                ...             ...                ...   
67509     commissariat_police-19031-01             NaN       commissariat   
67510                     aav-69387-01             NaN  ministere_justice   
67511                  mairie-83068-01             NaN                NaN   
67512                     cij-84092-01             NaN                NaN   
67513                  mairie-62715-01             NaN                NaN   

                telecopie                                                nom  \
service                                                                        
0        [05 59 52 40 37]  Antenne de Bayonne du service Pénitentiaire d'...   
1        [01 64 38 36 07]                          Mairie - Fontaine-le-Port   
2                     NaN                    Point-justice - France services   
3                     NaN  Service de renseignement en droit du travail -...   
4                     NaN    France Services - Nézignan-l’évêq

In [12]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 67514 entries, 0 to 67513
Data columns (total 38 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   plage_ouverture               54020 non-null  object
 1   site_internet                 47261 non-null  object
 2   copyright                     67514 non-null  object
 3   siren                         2620 non-null   object
 4   ancien_code_pivot             65852 non-null  object
 5   texte_reference               1326 non-null   object
 6   partenaire                    10518 non-null  object
 7   telecopie                     20583 non-null  object
 8   nom                           67514 non-null  object
 9   siret                         1373 non-null   object
 10  itm_identifiant               63071 non-null  object
 11  sigle                         139 non-null    object
 12  date_modification             67514 non-null  object
 13  adresse_courriel

In [17]:
display(data.notna().sum().sort_values(ascending=False))

categorie                       67514
copyright                       67514
statut_de_diffusion             67514
date_creation                   67514
nom                             67514
version_type                    67514
date_modification               67514
id                              67514
code_insee_commune              67389
adresse                         67360
telephone                       66970
pivot                           66166
ancien_code_pivot               65852
itm_identifiant                 63071
plage_ouverture                 54020
adresse_courriel                53121
site_internet                   47261
telecopie                       20583
partenaire_identifiant          10620
partenaire                      10518
formulaire_contact               7482
type_organisme                   5001
partenaire_date_modification     3906
commentaire_plage_ouverture      3216
sve                              3189
mission                          2994
siren       

In [13]:
display(data.loc[0])

plage_ouverture                                                               NaN
site_internet                                                                 NaN
copyright                                                 Ministère de la Justice
siren                                                                         NaN
ancien_code_pivot                                                   spip-64102-01
texte_reference                                                               NaN
partenaire                                                      ministere_justice
telecopie                                                        [05 59 52 40 37]
nom                             Antenne de Bayonne du service Pénitentiaire d'...
siret                                                                         NaN
itm_identifiant                                                               NaN
sigle                                                                         NaN
date_modificatio

Concentration sur la partie dense des données (colonnes pleines ou quasi (au moins 67380 valeurs)) :

In [18]:
dense_cols = [
    'categorie', 'copyright', 'statut_de_diffusion', 'date_creation',
    'nom', 'version_type', 'date_modification', 'id', 'code_insee_commune'
]

extract_1 = data[dense_cols]
display(extract_1)

categorie                                          copyright  \
service                                                                
0              SL                            Ministère de la Justice   
1              SL  Direction de l'information légale et administr...   
2              SL                            Ministère de la Justice   
3              SL  Direction de l'information légale et administr...   
4              SL  Direction de l'information légale et administr...   
...           ...                                                ...   
67509          SL  Direction de l'information légale et administr...   
67510          SL                            Ministère de la Justice   
67511          SL  Direction de l'information légale et administr...   
67512          SL  Direction de l'information légale et administr...   
67513          SL  Direction de l'information légale et administr...   

        statut_de_diffusion        date_creation  \
service                                            
0                      true  19/04/2022 19:52:06   
1                      true  18/01/2012 00:00:00   
2                      true  19/04/2022 19:51:40   
3                      true  17/09/2021 10:42:11   
4                      true  08/06/2021 10:25:30   
...                     ...                  ...   
67509                  true  18/01/2012 00:00:00   
67510                  true  19/04/2022 19:52:10   
67511                  true  18/01/2012 00:00:00   
67512                  true  18/01/2012 00:00:00   
67513                  true  18/01/2012 00:00:00   

                                                       nom version_type  \
service                                                                   
0        Antenne de Bayonne du service Pénitentiaire d'...    Publiable   
1                                Mairie - Fontaine-le-Port    Publiable   
2                          Point-justice - France services    Publiable   
3        Service de renseignement en droit du travail -...    Publiable   
4          France Services - Nézignan-l’évêque (itinérant)    Publiable   
...                                                    ...          ...   
67509         Commissariat de police de Brive-la-Gaillarde    Publiable   
67510       Association Le Mas Info droits victimes LE MAS    Publiable   
67511                                     Mairie - Grimaud    Publiable   
67512              Bureau information jeunesse - Le Pontet    Publiable   
67513                                   Mairie - Rocquigny    Publiable   

           date_modification                                    id  \
service                                                              
0        09/12/2022 23:12:45  0000774e-c649-4f14-855f-5fc83429b778   
1        08/03/2021 15:41:08  00007e4d-264c-43a0-b0b7-7f3b7dd995ab   
2        09/12/2022 23:12:11  0001a065-91eb-4532-a322-b705174eb492   
3        08/10/2021 09:04:48  0001e12f-03c3-42d1-b1b2-84cdccc7787a   
4        02/02/2022 14:05:33  0002884d-c249-408c-b460-9b26fdb86264   
...                      ...                                   ...   
67509    04/10/2022 15:41:06  fffba0d3-c386-4fba-a34c-6a693e8251ed   
67510    09/12/2022 23:12:50  fffcfc27-e705-4783-9529-0468255ea4b0   
67511    07/07/2021 17:50:18  fffdb221-ef56-4c28-b30e-601f60dae6d6   
67512    12/10/2020 15:38:32  ffff17c1-1203-4b4a-a405-aed8dabe6c70   
67513    23/06/2020 17:19:25  ffffff83-a9d2-4122-a272-72392164bada   

        code_insee_commune  
service                     
0                    64102  
1                    77188  
2                    45203  
3                    69266  
4                    34182  
...                    ...  
67509                19031  
67510                69387  
67511                83068  
67512                84092  
67513                62715  

[67514 rows x 9 columns]

Distributions empiriques :

In [21]:
for c in extract_1.columns:
    print(c, ':')
    display(extract_1[c].value_counts())

categorie :


SL     62365
SIL     5149
Name: categorie, dtype: int64

copyright :


Direction de l'information légale et administrative (Première ministre)    63609
Ministère de la Justice                                                     3905
Name: copyright, dtype: int64

statut_de_diffusion :


true    67514
Name: statut_de_diffusion, dtype: int64

date_creation :


18/01/2012 00:00:00    47624
30/06/2008 00:00:00     2147
19/04/2022 19:52:45       39
19/04/2022 19:52:42       39
19/04/2022 19:52:20       38
                       ...  
27/04/2021 14:25:00        1
31/05/2022 18:16:24        1
03/03/2022 11:44:19        1
24/01/2019 10:40:19        1
26/11/2018 11:27:00        1
Name: date_creation, Length: 11582, dtype: int64

nom :


Point-justice - France services             386
Mairie - Sainte-Colombe                      12
Pôle Politiques publiques                    12
Mairie - Saint-Sauveur                       11
Pôle Modernisation et moyens                 10
                                           ... 
Brigade de gendarmerie - Mussy-sur-Seine      1
Mairie - Raissac-d'Aude                       1
Mairie - Ligny-le-Ribault                     1
Mairie - Lauroux                              1
Bureau information jeunesse - Le Pontet       1
Name: nom, Length: 64728, dtype: int64

version_type :


Publiable    67514
Name: version_type, dtype: int64

date_modification :


16/09/2022 10:17:08    42
30/09/2022 21:57:01    33
16/05/2018 11:30:49    33
30/09/2022 21:57:04    33
30/09/2022 21:57:03    33
                       ..
25/01/2021 09:03:03     1
25/04/2022 14:34:15     1
10/08/2021 15:08:46     1
16/04/2021 10:49:10     1
23/06/2020 17:19:25     1
Name: date_modification, Length: 37844, dtype: int64

id :


0000774e-c649-4f14-855f-5fc83429b778    1
a9d72538-df2d-4220-bf57-9733f602aef4    1
a9da07c5-af78-4966-9ab7-436f4a02c6fd    1
a9da45da-ff4d-41dd-b1aa-1cd90e783f70    1
a9da68b6-ed85-4ebe-89dd-7db6da3ac0b1    1
                                       ..
54fa695e-1906-4ea5-a5d7-3247b142cf76    1
54fb9d64-9987-4e15-8b16-087bbdb96520    1
54fcfb14-d43c-4a5b-b830-7db91554ab06    1
54fd4f42-9749-4101-a12e-7e1c84a99f1e    1
ffffff83-a9d2-4122-a272-72392164bada    1
Name: id, Length: 67514, dtype: int64

code_insee_commune :


31555    229
35238    200
59350    195
33063    194
44109    177
        ... 
42288      1
84021      1
72182      1
25514      1
62715      1
Name: code_insee_commune, Length: 35147, dtype: int64

On peut :
* supprimer les colonnes constantes :
    * `statut_de_diffusion` qui est toujours `true`
    * `version_type` qui est toujours `Publiable`
* passer la colonne `id` en index
* coder les deux valeurs possibles de `copyright` en DILA et MJ

NB :
* 35147 communes (codes INSEE) sont représentées
* cf. [Spec](https://echanges.dila.gouv.fr/OPENDATA/Base_donn%c3%a9es_locales/Specifications-datagouv-base-de-donnees-locales-Service-public_V1.1.pdf) SL vs. SIL dénotent "Service Local" vs. "Service Institutionnel Local"
* les types de service sont compris dans les noms, dont certains sont composés : [Type] - [Collectivité]

In [23]:
dense_cols = [
    'categorie', 'copyright', 'date_creation',
    'nom', 'date_modification', 'id', 'code_insee_commune'
]

extract_1 = data[dense_cols]
extract_1 = extract_1.set_index('id')
extract_1.copyright = extract_1.copyright.apply(
    lambda x: 'MJ' if x == 'Ministère de la Justice' else 'DILA'
)
display(extract_1)

categorie copyright        date_creation  \
id                                                                              
0000774e-c649-4f14-855f-5fc83429b778        SL        MJ  19/04/2022 19:52:06   
00007e4d-264c-43a0-b0b7-7f3b7dd995ab        SL      DILA  18/01/2012 00:00:00   
0001a065-91eb-4532-a322-b705174eb492        SL        MJ  19/04/2022 19:51:40   
0001e12f-03c3-42d1-b1b2-84cdccc7787a        SL      DILA  17/09/2021 10:42:11   
0002884d-c249-408c-b460-9b26fdb86264        SL      DILA  08/06/2021 10:25:30   
...                                        ...       ...                  ...   
fffba0d3-c386-4fba-a34c-6a693e8251ed        SL      DILA  18/01/2012 00:00:00   
fffcfc27-e705-4783-9529-0468255ea4b0        SL        MJ  19/04/2022 19:52:10   
fffdb221-ef56-4c28-b30e-601f60dae6d6        SL      DILA  18/01/2012 00:00:00   
ffff17c1-1203-4b4a-a405-aed8dabe6c70        SL      DILA  18/01/2012 00:00:00   
ffffff83-a9d2-4122-a272-72392164bada        SL      DILA  18/01/2012 00:00:00   

                                                                                    nom  \
id                                                                                        
0000774e-c649-4f14-855f-5fc83429b778  Antenne de Bayonne du service Pénitentiaire d'...   
00007e4d-264c-43a0-b0b7-7f3b7dd995ab                          Mairie - Fontaine-le-Port   
0001a065-91eb-4532-a322-b705174eb492                    Point-justice - France services   
0001e12f-03c3-42d1-b1b2-84cdccc7787a  Service de renseignement en droit du travail -...   
0002884d-c249-408c-b460-9b26fdb86264    France Services - Nézignan-l’évêque (itinérant)   
...                                                                                 ...   
fffba0d3-c386-4fba-a34c-6a693e8251ed       Commissariat de police de Brive-la-Gaillarde   
fffcfc27-e705-4783-9529-0468255ea4b0     Association Le Mas Info droits victimes LE MAS   
fffdb221-ef56-4c28-b30e-601f60dae6d6                                   Mairie - Grimaud   
ffff17c1-1203-4b4a-a405-aed8dabe6c70            Bureau information jeunesse - Le Pontet   
ffffff83-a9d2-4122-a272-72392164bada                                 Mairie - Rocquigny   

                                        date_modification code_insee_commune  
id                                                                            
0000774e-c649-4f14-855f-5fc83429b778  09/12/2022 23:12:45              64102  
00007e4d-264c-43a0-b0b7-7f3b7dd995ab  08/03/2021 15:41:08              77188  
0001a065-91eb-4532-a322-b705174eb492  09/12/2022 23:12:11              45203  
0001e12f-03c3-42d1-b1b2-84cdccc7787a  08/10/2021 09:04:48              69266  
0002884d-c249-408c-b460-9b26fdb86264  02/02/2022 14:05:33              34182  
...                                                   ...                ...  
fffba0d3-c386-4fba-a34c-6a693e8251ed  04/10/2022 15:41:06              19031  
fffcfc27-e705-4783-9529-0468255ea4b0  09/12/2022 23:12:50              69387  
fffdb221-ef56-4c28-b30e-601f60dae6d6  07/07/2021 17:50:18              83068  
ffff17c1-1203-4b4a-a405-aed8dabe6c70  12/10/2020 15:38:32              84092  
ffffff83-a9d2-4122-a272-72392164bada  23/06/2020 17:19:25              62715  

[67514 rows x 6 columns]

Exemple d'extraction de sous-ensemble : les mairies :

In [24]:
extract_2 = extract_1[extract_1.nom.str.startswith('Mairie - ')]
display(extract_2)

categorie copyright        date_creation  \
id                                                                              
00007e4d-264c-43a0-b0b7-7f3b7dd995ab        SL      DILA  18/01/2012 00:00:00   
00033429-1b83-46b4-9101-3a2ad178af79        SL      DILA  18/01/2012 00:00:00   
000429a7-4ce5-4173-8f7e-80da7c8e7702        SL      DILA  18/01/2012 00:00:00   
00053c80-6183-4252-a5dd-b4831440c7d0        SL      DILA  18/01/2012 00:00:00   
0007006a-18a1-41cc-b5d8-adea0962070e        SL      DILA  18/01/2012 00:00:00   
...                                        ...       ...                  ...   
fff78c88-87c3-4dc9-8bab-8031b7b99318        SL      DILA  18/01/2012 00:00:00   
fff7bd00-9c35-4422-8c13-76162b5c0f12        SL      DILA  18/01/2012 00:00:00   
fffac32f-d889-4d70-ab56-1a3127ab922f        SL      DILA  18/01/2012 00:00:00   
fffdb221-ef56-4c28-b30e-601f60dae6d6        SL      DILA  18/01/2012 00:00:00   
ffffff83-a9d2-4122-a272-72392164bada        SL      DILA  18/01/2012 00:00:00   

                                                                          nom  \
id                                                                              
00007e4d-264c-43a0-b0b7-7f3b7dd995ab                Mairie - Fontaine-le-Port   
00033429-1b83-46b4-9101-3a2ad178af79             Mairie - Courcelles-sur-Aire   
000429a7-4ce5-4173-8f7e-80da7c8e7702             Mairie - Villaines-la-Gonais   
00053c80-6183-4252-a5dd-b4831440c7d0      Mairie - Paris - 13e arrondissement   
0007006a-18a1-41cc-b5d8-adea0962070e  Mairie - Coise-Saint-Jean-Pied-Gauthier   
...                                                                       ...   
fff78c88-87c3-4dc9-8bab-8031b7b99318             Mairie - Puilly-et-Charbeaux   
fff7bd00-9c35-4422-8c13-76162b5c0f12                          Mairie - Teyjat   
fffac32f-d889-4d70-ab56-1a3127ab922f                        Mairie - Gillaumé   
fffdb221-ef56-4c28-b30e-601f60dae6d6                         Mairie - Grimaud   
ffffff83-a9d2-4122-a272-72392164bada                       Mairie - Rocquigny   

                                        date_modification code_insee_commune  
id                                                                            
00007e4d-264c-43a0-b0b7-7f3b7dd995ab  08/03/2021 15:41:08              77188  
00033429-1b83-46b4-9101-3a2ad178af79  22/06/2018 14:47:06              55128  
000429a7-4ce5-4173-8f7e-80da7c8e7702  10/08/2021 16:36:23              72375  
00053c80-6183-4252-a5dd-b4831440c7d0  04/10/2022 11:34:45              75113  
0007006a-18a1-41cc-b5d8-adea0962070e  03/03/2021 08:20:09              73089  
...                                                   ...                ...  
fff78c88-87c3-4dc9-8bab-8031b7b99318  16/07/2020 09:10:44              08347  
fff7bd00-9c35-4422-8c13-76162b5c0f12  09/11/2022 15:47:22              24548  
fffac32f-d889-4d70-ab56-1a3127ab922f  09/07/2021 15:19:33              52222  
fffdb221-ef56-4c28-b30e-601f60dae6d6  07/07/2021 17:50:18              83068  
ffffff83-a9d2-4122-a272-72392164bada  23/06/2020 17:19:25              62715  

[35105 rows x 6 columns]

Exemple 2 : seulement les mairies du Vaucluse

In [25]:
extract_3 = extract_2[extract_2.code_insee_commune.str.startswith('84')]
display(extract_3)

categorie copyright        date_creation  \
id                                                                              
0832bdb9-313d-4974-96b3-562b845f40f7        SL      DILA  18/01/2012 00:00:00   
088685f7-de42-4846-ac7e-a4d4fc6cd902        SL      DILA  18/01/2012 00:00:00   
0a479367-8084-4a1a-88f4-36efd23744e5        SL      DILA  18/01/2012 00:00:00   
0a4c3010-d46d-4f1c-b075-c0abd5ae42df        SL      DILA  18/01/2012 00:00:00   
0a61a07d-2aba-429c-ae8d-cff919bc476b        SL      DILA  18/01/2012 00:00:00   
...                                        ...       ...                  ...   
f9921983-4229-439b-88a0-184351282d56        SL      DILA  18/01/2012 00:00:00   
fad6a7a6-9b8d-4292-9d59-b762e7ff30e6        SL      DILA  18/01/2012 00:00:00   
faf31ac4-3f51-48f1-a53b-cee765a3332e        SL      DILA  18/01/2012 00:00:00   
fba2f768-6d90-4892-8e64-e2d5cd82defa        SL      DILA  18/01/2012 00:00:00   
fc099270-58ea-45ce-8fc8-474b6961df62        SL      DILA  18/01/2012 00:00:00   

                                                                      nom  \
id                                                                          
0832bdb9-313d-4974-96b3-562b845f40f7           Mairie - La Motte-d'Aigues   
088685f7-de42-4846-ac7e-a4d4fc6cd902                       Mairie - Lioux   
0a479367-8084-4a1a-88f4-36efd23744e5  Mairie - Saint-Marcellin-lès-Vaison   
0a4c3010-d46d-4f1c-b075-c0abd5ae42df                         Mairie - Apt   
0a61a07d-2aba-429c-ae8d-cff919bc476b                  Mairie - Le Barroux   
...                                                                   ...   
f9921983-4229-439b-88a0-184351282d56                Mairie - Richerenches   
fad6a7a6-9b8d-4292-9d59-b762e7ff30e6        Mairie - Morières-lès-Avignon   
faf31ac4-3f51-48f1-a53b-cee765a3332e            Mairie - Crillon-le-Brave   
fba2f768-6d90-4892-8e64-e2d5cd82defa                     Mairie - Pertuis   
fc099270-58ea-45ce-8fc8-474b6961df62               Mairie - Lagarde-d'Apt   

                                        date_modification code_insee_commune  
id                                                                            
0832bdb9-313d-4974-96b3-562b845f40f7  07/07/2021 17:52:37              84084  
088685f7-de42-4846-ac7e-a4d4fc6cd902  07/07/2021 17:53:25              84066  
0a479367-8084-4a1a-88f4-36efd23744e5  07/07/2021 17:59:49              84111  
0a4c3010-d46d-4f1c-b075-c0abd5ae42df  27/07/2021 14:50:24              84003  
0a61a07d-2aba-429c-ae8d-cff919bc476b  07/07/2021 17:51:28              84008  
...                                                   ...                ...  
f9921983-4229-439b-88a0-184351282d56  07/07/2021 18:00:52              84097  
fad6a7a6-9b8d-4292-9d59-b762e7ff30e6  07/07/2021 17:53:33              84081  
faf31ac4-3f51-48f1-a53b-cee765a3332e  07/07/2021 17:42:40              84041  
fba2f768-6d90-4892-8e64-e2d5cd82defa  07/07/2021 17:52:53              84089  
fc099270-58ea-45ce-8fc8-474b6961df62  07/07/2021 17:51:27              84060  

[151 rows x 6 columns]

Next step : partant d'une liste de communes (codes INSEE), ici celles du Vaucluse, charger les fichiers de services locaux de ces communes.